plugins/generator-1.17.1/forge-1.17.1/templates/particle.java.ftl

```diff
@@ -0,0 +1,138 @@
+<#--
+ # MCreator (https://mcreator.net/)
+ # Copyright (C) 2012-2020, Pylo
+ # Copyright (C) 2020-2021, Pylo, opensource contributors
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+<#include "procedures.java.ftl">
+
+package ${package}.particle;
+
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD) public class ${name}Particle {
+
+	public static final SimpleParticleType particle = new SimpleParticleType(${data.alwaysShow});
+
+	@SubscribeEvent public static void registerParticleType(RegistryEvent.Register<ParticleType<?>> event) {
+		event.getRegistry().register(particle.setRegistryName("${registryname}"));
+	}
+
+	@OnlyIn(Dist.CLIENT) @SubscribeEvent public static void registerParticle(ParticleFactoryRegisterEvent event) {

```

Comment:
You should remove registration here if you do it in the common registry

Reply:
The thing you pointed registers particle factory (provider since 1.17.x) that is inner particle class, but `registerParticleType()` method can be moved instead.

Reply:
:heavy_check_mark: Done
P. S. I need your help with package name [:arrow_up:](https://github.com/MCreator/MCreator/pull/1729#discussion_r705476433)

Reply:
This is still here and not in the global common registry

Reply:
Working on it.